ebaysdk/inventorymanagement/__init__.py

```diff
@@ -0,0 +1,259 @@
+                           config_file=kwargs.get('config_file', 'ebay.yaml'))
+
+        # override yaml defaults with args sent to the constructor
+        self.config.set('domain', kwargs.get('domain', 'api.ebay.com'))
+        self.config.set('uri', '/selling/inventory/v1')
+        self.config.set('https', True)
+        self.config.set('warnings', True)
+        self.config.set('errors', True)
+        self.config.set('siteid', None)
+        self.config.set('response_encoding', 'XML')
+        self.config.set('request_encoding', 'XML')
+        self.config.set('proxy_host', None)
+        self.config.set('proxy_port', None)
+        self.config.set('token', None)
+        self.config.set('iaf_token', None)
+        self.config.set('appid', None)
+        self.config.set('version', '1.0.0')
+        self.config.set('service', 'InventoryManagement')
+        self.config.set('doc_url', 'http://developer.ebay.com/Devzone/store-pickup/InventoryManagement/index.html')
+
+        self.datetime_nodes = ['starttimefrom', 'timestamp', 'starttime',
+                               'endtime']
+        self.base_list_nodes = [
+        ]
+
+    endpoints = {
+        'addinventorylocation': 'locations/delta/add',
+        'addinventory': 'inventory/delta/add',
+        'deleteinventory': 'inventory/delta/delete',
+        'deleteinventorylocation': 'locations/delta/delete',
+    }
+
+    def build_request_url(self, verb):
+        url = super(Connection, self).build_request_url(verb)
+        endpoint = self.endpoints[verb.lower()]
+        return "{0}/{1}".format(url, endpoint)
+
+    def build_request_headers(self, verb):
+        return {
+            "Authorization": "TOKEN {0}".format(self.config.get('token')),
+            "Content-Type": "application/xml"
+        }
+
+    def build_request_data(self, verb, data, verb_attrs):
+        xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?>"
+        xml += "<" + verb + "Request>"
+        xml += dict2xml(data)
+        xml += "</" + verb + "Request>"
+
+        return xml
+
+    def warnings(self):
+        warning_string = ""
+
+        if len(self._resp_body_warnings) > 0:
+            warning_string = "%s: %s" \
+                % (self.verb, ", ".join(self._resp_body_warnings))
+
+        return warning_string
+
+
+    def _get_resp_body_errors(self):
+        """Parses the response content to pull errors.
+
+        Child classes should override this method based on what the errors in the
+        XML response body look like. They can choose to look at the 'ack',
+        'Errors', 'errorMessage' or whatever other fields the service returns.
+        the implementation below is the original code that was part of error()
+        """
+
+        if self._resp_body_errors and len(self._resp_body_errors) > 0:
+            return self._resp_body_errors
+
+        errors = []
+        warnings = []
+        resp_codes = []
+
+        if self.verb is None:
+            return errors
```

Comment:
You mean `format` instead of `modulo operator`? To be honest, I just copied it from other class, it works perfectly because there is no input here so it will never have utf-8 related bugs as I had in trading api.